app/src/main/java/com/oracle/javafx/scenebuilder/app/DocumentWindowController.java

```diff
@@ -1573,10 +1574,25 @@ private void performPaste() {
         } else {
             assert isTextInputControlEditing(focusOwner);
             final TextInputControl tic = getTextInputControl(focusOwner);
-            tic.paste();
+            performTextInputControlPaste(tic);
```

Comment:
Since the fix is so simple now, and given the fact that it should be removed once the JBS issue gets fixed, I don't think we need a full class just for it?

It should be as simple as:

```
final TextInputControl tic = getTextInputControl(focusOwner);
// TODO: Once the issue https://bugs.openjdk.java.net/browse/JDK-8205915 is solved, 
// this if expression should be removed.
if (!EditorPlatform.IS_MAC) {
    tic.paste();
}
```

Reply:
Will this PR merge with JavaFX 18?

I decided for a separate class just for testability. But you're right, let me rework this.

Reply:
So, its now much simpler - we kind of need to keep track of this. As soon as [openjdk/jfx#704](https://github.com/openjdk/jfx/pull/704 ) is merged, the paste operation in Scene Builder will be broken on MacOS.

Reply:
Indeed, I can't say when that PR will be merged. Most certainly it won't make it to 18 (jfx18 stabilisation branch scheduled for this week), and SB18 will still need this patch. In the end, we might need some JavaFX version check to apply it or not...

Reply:
Okay, my proposal here is, to introduce a test which ensures, that this workaround is only applied to Scene Builder 18. If any version number is introduced, there will be a failing unit test with a reminder not towards the related PR.

What do you think? It is still some time till release 19 and a failing test certainly will be a good reminder.